lib/gitsh/tab_completion/dsl/maybe_operation_factory.rb

```diff
@@ -0,0 +1,19 @@
+module Gitsh
+  module TabCompletion
+    module DSL
+      class MaybeOperationFactory
+        attr_reader :child
```

Comment:
I exposed them for tests, which isn't great, but the parser specs would be significantly more complex if I didn't.